MapboxNavigation/Route.swift

```diff
@@ -0,0 +1,41 @@
+import MapboxDirections
+import Turf
+
+extension Route {
+    /**
+     Returns a polyline extending a given distance in either direction from a given maneuver along the route.
+     
+     The maneuver is identified by a leg index and step index, in case the route doubles back on itself.
+     
+     - parameter legIndex: Zero-based index of the leg containing the maneuver.
+     - parameter stepIndex: Zero-based index of the step containing the maneuver.
+     - parameter distance: Distance by which the resulting polyline extends in either direction from the maneuver.
+     - returns: A polyline whose length is twice `distance` and whose centroid is located at the maneuver.
+     */
+    func polylineAroundManeuver(legIndex: Int, stepIndex: Int, distance: CLLocationDistance) -> Polyline {
```

Comment:
This looks good but isn't it still technically possible to have a step which could double back on itself?
I haven't really wrapped my head around this yet but why are we trimming instead of slicing? Trimming takes all coordinates within a radius, right? It's just a bit unlikely that a double backed part of the polyline wouldn't be another step. I think there's a single step which double backs on itself in Weehawken New Jersey, from-to `-74.01701276685586,40.76621002800809;-74.02209890200525,40.76787957422633`

If the assumption above is correct, do you think it would be possible to make some changes to the bottom part of this function to use `LineString.sliced(from:to:)` instead of `LineString.trimmed(from:distance:)` and rename `polylineAroundManeuver` to `polylineAlongManeuver`?

Reply:
> This looks really good but isn't it still technically possible to have a step which could double back on itself?

It is true that a single step can double back on itself, as in some of the cases mentioned in #1733. This also concerned me at first. Fortunately, a step can only have one maneuver, and we know for sure the position of that maneuver within the step’s coordinate array (the first index). So as long as the partition occurs at the right place, a step that doubles back on itself will still result in a half-shaft polyline that begins at the maneuver and extends the correct distance in either direction.

> do you think it would be possible to make some changes to the bottom part of this function to use `LineString.sliced(from:to:)` instead of `LineString.trimmed(from:distance:)`

By accepting coordinates as arguments, `LineString.sliced(from:to:)` assumes the polyline never doubles back on itself. Perhaps Turf should provide an indexed alternative to this method, along the lines of mapbox/turf-swift#67.